tests/shared.py

```diff
@@ -127,19 +130,27 @@ def do_entity_test(
             # save the updated RO-Crate metadata
             with open(temp_rocrate_path / "ro-crate-metadata.json", "w") as f:
                 json.dump(rocrate, f)
-        # update the RO-Crate metadata using RDF triples, if required
-        if rocrate_entity_mod_function is not None:
-            rocrate_graph = load_graph_and_preserve_relative_ids(rocrate)
-            rocrate_graph = rocrate_entity_mod_function(rocrate_graph)
+        # update the RO-Crate metadata using SPARQL, if required
+        if rocrate_entity_mod_sparql is not None:
+            base = "https://example.org/"
+            rocrate_graph = rdflib.Graph()
+            rocrate_graph.parse(data=rocrate, format="json-ld", publicID=base)
+
+            query = f"""BASE <{base}>
+{rocrate_entity_mod_sparql}
+"""
+            rocrate_graph.update(query)
             # save the updated RO-Crate metadata
             context = "https://w3id.org/ro/crate/1.1/context"
             rocrate_graph.serialize(
                 Path(temp_rocrate_path, "ro-crate-metadata.json"),
                 format="json-ld",
                 context=context,
                 indent=2,
-                use_native_types=True
+                use_native_types=True,
+                base=base,
             )
+            # TODO: tweak the generated crate to strip leading slash from relative URIs, and ensure RDE ID is ./
```

Comment:
This might not actually pose a real issue, because I don't think the validator will really care. It's just slightly fragile. I'm not sure if selecting specific ids with SPARQL will work, for example (could be a problem for RDE checks, actually)